### DISCUSSION TITLE
trainer doesn't save any checkpoint when using  checkpoint_callback

### DISCUSSION THREAD

**Discussion by allenhung1025** [Lightning-AI/pytorch-lightning#12929] [General]
### Environment
* OS: Ubuntu 18.04.5 LTS
* python3.7.11
* pytorch 1.11.0+cu102
* pytorch lightning 1.6.1

### Code
* validation_step
```python
def validation_step(self, batch, batch_idx):
        data, aug_data, label = batch
        input_ids, attention_mask, labels = data['input_ids'], data['attention_mask'], label
        aug_input_ids, aug_attention_mask = aug_data['input_ids'], aug_data['attention_mask']
        input_ids, attention_mask, labels = input_ids.cuda(), attention_mask.cuda(), labels.cuda()
        aug_input_ids, aug_attention_mask = aug_input_ids.cuda(), aug_attention_mask.cuda()

        output = self.compute_loss_prediction_and_metrics(input_ids, aug_input_ids, attention_mask, aug_attention_mask, labels)
        loss, contrastive_loss, poincare_loss, valid_pred, valid_label, valid_acc, valid_micro_f1, valid_macro_f1, valid_weighted_f1 = output

        self.valid_pred.extend(valid_pred)
        self.valid_label.extend(valid_label)

        self.log('valid_acc', valid_acc, on_step = False, on_epoch = True)
        self.log('valid_micro_f1', valid_micro_f1)
        self.log('valid_macro_f1', valid_macro_f1)
        self.log('valid_weighted_f1', valid_weighted_f1)
        
        return {'valid_loss': loss, 'valid_pred': valid_pred, 'valid_acc': valid_acc,
                'valid_micro_f1': valid_micro_f1, 'valid_macro_f1': valid_macro_f1, 'valid_weighted_f1': valid_weighted_f1}
```
* validation_epoch_end
```python
def validation_epoch_end(self , outs):
        valid_acc, valid_micro_f1, valid_macro_f1, valid_weighted_f1 = self.acc_and_f1_score(self.valid_pred, self.valid_label)
        self.valid_pred = []
        self.valid_label = []
        self.log('valid_acc_epoch_end', valid_acc)
        self.log('valid_micro_f1_epoch_end', valid_micro_f1)
        self.log('valid_macro_f1_epoch_end', valid_macro_f1)
        self.log('valid_weighted_f1_epoch_end', valid_weighted_f1)

        return {'val_loss': outs[0]['valid_loss'], 'valid_acc_epoch_end': valid_acc, 'valid_micro_f1_epoch_end':valid_micro_f1,
                'valid_macro_f1_epoch_end': valid_macro_f1, 'valid_weighted_f1_epoch_end':valid_weighted_f1}
```
* trainer 
```python
checkpoint_callback = ModelCheckpoint(dirpath = args.save_path, 
                                      verbose=True,
                                      save_top_k = -1,
                                      save_last=True)
 trainer = pl.Trainer(gpus = 1,
                     callbacks = [checkpoint_callback],
                     logger=logger,
                     default_root_dir=args.save_path,
                     max_epochs = args.max_epochs)
 trainer.fit(gm)
```
### Expected behavior
* save a model for each epoch under args.save_path

### Real behavior
* No model is saved under args.save_path
* only log the loss, accuracy, ..., etc.
![Screen Shot 2022-04-29 at 4 06 13 PM](https://user-images.githubusercontent.com/22190213/165906915-faa9b61e-230b-49db-a67b-9aef0c1926b2.png)

### Request for help
* Does anyone can point me out where I got wrong? I have searched on Google, looked the document thoroughly, and still couldn't find a solution. Thanks in advance.

**Comment by rohitgr7** [comment]
can you check the actual dirpath of the checkpoint callback using `trainer.checkpoint_callback.dirpath` and see if there are any checkpoints there?

if they are still now showing up, can you share a reproducible minimal example using [BoringModel](https://colab.research.google.com/github/PytorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report/bug_report_model.ipynb)?

**Reply by allenhung1025** [reply]
Hi @rohitgr7, 

When I use [BoringMode](https://colab.research.google.com/github/PytorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report/bug_report_model.ipynb) with model_checkpoint and trainer that I defined, the tariner can save the checkpoint successfully. 
<img width="675" alt="Screen Shot 2022-05-03 at 9 40 03 AM" src="https://user-images.githubusercontent.com/22190213/166393133-a7f7e5ab-bcc5-49e7-952b-8a3a02622884.png">

However, when I switched back to the model I defined, the trainer wouldn't save any checkpoint under `trainer.checkpoint_callback.dirpath`. The logging worked fine, though.
<img width="474" alt="Screen Shot 2022-05-03 at 9 53 17 AM" src="https://user-images.githubusercontent.com/22190213/166393931-cffa3849-0e87-49bd-a18e-a52a844d6599.png">

<img width="551" alt="Screen Shot 2022-05-03 at 9 48 18 AM" src="https://user-images.githubusercontent.com/22190213/166393634-63cf5933-9910-42d6-a819-549f4a66a618.png">


Thanks for your kind reply.

**Reply by rohitgr7** [reply]
```py
checkpoint_callback = ModelCheckpoint(dirpath = args.save_path, 
                                      verbose=True,
                                      save_top_k = -1,
                                      save_last=True)
```
since you are not using any monitor, it should just checkpoint after every epoch.

do you see any warnings being emitted from checkpoint callback during the run?

**Reply by allenhung1025** [reply]
The cloest warning related to callback is stating that the directory exists.
``` 
/home/allenhung/miniconda3/envs/finegrained/lib/python3.7/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:608: UserWarning: Checkpoint directory /home/allenhung/fine-grained-classification-$
epro/checkpoint_GE_4 exists and is not empty.
  rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
  ```

**Reply by rohitgr7** [reply]
so it's showing the directory is not empty and not even storing anything in that. this is weird.
not sure if there is some issue in your file-system, but I'd need a reproducible script to check the issue.
